src/services/dispatchers.ts

```diff
@@ -0,0 +1,111 @@
+import { readFile } from 'fs/promises';
+import { Dispatcher, DispatcherCreation, OAuth1Client, rqlBuilder } from '@extrahorizon/javascript-sdk';
+import { blue, green, yellow } from 'chalk';
+import * as dispatcherRepository from '../repositories/dispatchers';
+
+const cliManagedTag = 'EXH_CLI_MANAGED';
+
+export async function sync(sdk: OAuth1Client, file: string) {
+  // TODO: Add permission check for current user?
+  console.log(yellow(`Synchronizing Dispatchers from ${file}`));
+
+  const localDispatchers = await extractDispatchersFromFile(file);
+
+  const rql = rqlBuilder().eq('tags', cliManagedTag).build();
+  const exhDispatchers = await dispatcherRepository.findAll(sdk, rql);
+
+  for (const localDispatcher of localDispatchers) {
+    console.group(blue(`Synchronizing Dispatcher: ${localDispatcher.name}`));
+
+    // Ensure all Dispatchers and actions have name fields
+    assertRequiredFields(localDispatcher);
+
+    // TODO: This does not account for dispatchers that exist in Extra Horizon with an existing name, but not a EXH_CLI_MANAGED tag
+    const exhDispatcher = exhDispatchers.find(({ name }) => name === localDispatcher.name);
+    await synchronizeDispatcher(sdk, localDispatcher, exhDispatcher);
+    console.groupEnd();
+  }
+}
+
+async function synchronizeDispatcher(sdk: OAuth1Client, localDispatcher: DispatcherCreation, exhDispatcher: Dispatcher | undefined) {
```

Comment:
(minor/stupid): `exhDispatcher?: Dispatcher` ?

Reply:
So actually kind of interesing, DispatcherCreation is correct. Your locally stored objects aren't full Dispatchers nor' ever will be. They're always going to be what is used to create a Dispatcher.

No id and timestamps etc...